### PR TITLE
Allow deployemnts without genesis/audit, extending existing network

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: src/deploy.rs [hunk 1]
```diff
@@ -78,27 +87,39 @@ impl DeployCmd {
...
-        })?;
-        println!("Obtained multiaddr for genesis node: {genesis_multiaddr}");
+            contact.clone()
+        } else {
+            self.print_ansible_run_banner(n, total, "Provision Genesis Node");
+            self.provision_genesis_node().await.map_err(|err| {
+                println!("Failed to provision genesis node {err:?}");
+                err
+            })?;
+            n += 1;
```
Comment (RolandSherwin) [inline src/deploy.rs:102]: A tiny nit pick, but if `!is_fresh_network` then `total` has to be updated. It just prints the max steps in the banner.

File: src/deploy.rs [hunk 2]
```diff
@@ -60,7 +67,9 @@ impl DeployCmd {
                 BinaryOption::Versioned { .. } => false,
             }
         };
-        self.create_infra(build_custom_binaries)
+
+        let is_fresh_network = self.bootstrap_peer.is_none();
```
Comment (RolandSherwin) [inline src/deploy.rs:71]: Got two variables doing the same thing here `is_fresh_network` and `this_is_a_new_network`

File: src/deploy.rs (hunk 1 above)
Comment (joshuef) [reply src/deploy.rs:102]: i'm not sure what you mean to change here.

Comment (joshuef) [reply src/deploy.rs:102]: the total is being updated, no? with `n+=1` ?

**Pull request merged**